src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -209,6 +218,24 @@ public static class PackageDownloader
                     sourceRepository.PackageSource.Source));
             }
 
+            if (result.Type == DownloadResourceResultType.Cancelled)
+            {
+                throw new FatalProtocolException(String.Format(
```

Comment:
this shouldn't be a fatalprocotolexception because this is actually a success case